chains/solana/contracts/programs/ccip-router/src/token_context.rs

```diff
@@ -23,47 +23,50 @@ pub struct TokenAdminRegistry {
 }
 
 #[derive(Accounts)]
-#[instruction(mint: Pubkey)]
-pub struct RegisterTokenAdminRegistryViaGetCCIPAdmin<'info> {
+pub struct RegisterTokenAdminRegistryByCCIPAdmin<'info> {
     #[account(
         seeds = [CONFIG_SEED],
         bump,
         constraint = valid_version(config.load()?.version, MAX_CONFIG_V) @ CcipRouterError::InvalidInputs,
     )]
     pub config: AccountLoader<'info, Config>,
     #[account(
-        init,
-        seeds = [TOKEN_ADMIN_REGISTRY_SEED, mint.as_ref()],
+        init, // TODO: change to init_if_needed
+        seeds = [TOKEN_ADMIN_REGISTRY_SEED, mint.key().as_ref()],
         bump,
         payer = authority,
         space = ANCHOR_DISCRIMINATOR + TokenAdminRegistry::INIT_SPACE,
         constraint = uninitialized(token_admin_registry.version) @ CcipRouterError::InvalidInputs,
     )]
     pub token_admin_registry: Account<'info, TokenAdminRegistry>,
+    pub mint: InterfaceAccount<'info, Mint>, // underlying token that the pool wraps
+    // The following validation is the only difference between the two contexts
+    // Only CCIP Admin can propose an owner for the token admin registry if not the mint authority
     #[account(mut, address = config.load()?.owner @ CcipRouterError::Unauthorized)]
     pub authority: Signer<'info>,
     pub system_program: Program<'info, System>,
 }
 
 #[derive(Accounts)]
-pub struct RegisterTokenAdminRegistryViaOwner<'info> {
+pub struct RegisterTokenAdminRegistryByOwner<'info> {
     #[account(
         seeds = [CONFIG_SEED],
         bump,
         constraint = valid_version(config.load()?.version, MAX_CONFIG_V) @ CcipRouterError::InvalidInputs,
     )]
     pub config: AccountLoader<'info, Config>,
     #[account(
-        init,
+        init, // TODO: change to init_if_needed
```

Comment:
If the purpose of changing to `init_if_needed` is to be able to override a pending administrator, would it be a bit cleaner to just implement a `override_pending_administrator` instruction? WDYT?

I read that `init_if_needed` is a security concern, and I do think it results in slightly more confusing flow than the alternative of well delimited initialization and "editing" instructions, but I don't know if there are any constraints that I'm not seeing.

Reply:
Mmm, I guess you are right. I tried to keep it as similar as possible to EVM, that's why I thought of using an `init_if_needed`, but as this functionality should be used only to fix a missconfiguration it shouldn't be part of the common use case. It's safer and more clear to have a separate `override_pending_administrator` message.